multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1773,15 +1773,8 @@ def _default_predict(
                     else:
                         outputs = pred_writer.collect_all_gpu_results(num_gpus=num_gpus)
                 elif self._problem_type == OBJECT_DETECTION:
-                    # reformat single gpu output for object detection
-                    # outputs shape: num_batch, 1(["bbox"]), batch_size, 80, n, 5
-                    # output LABEL if exists for evaluations
-                    outputs = [
-                        {BBOX: bbox, LABEL: ele[LABEL][i]} if LABEL in ele else {BBOX: bbox}
-                        for ele in outputs
-                        for i, bbox in enumerate(ele[BBOX])
-                    ]
-
+                    # Unpack outputs for object detection while using single gpu
+                    outputs = [output for batch_outputs in outputs for output in batch_outputs]
```

Comment:
This list comprehension logic looks problematic.

Reply:
This is simply unpacking a list with two dimensions (`<num_batches>` batches of `<batch_size>` samples each).